pymc/model.py

```diff
@@ -1728,7 +1728,7 @@ def point_logps(self, point=None, round_vals=2):
 Model._context_class = Model
 
 
-def set_data(new_data, model=None):
+def set_data(new_data, coords,model=None):
```

Comment:
Don't change the signature
```suggestion
def set_data(new_data, model=None, *, coords=None):
```
Also please make sure to run `pre-commit run --all` before uploading to avoid unnecessary CI runs due to formatting problems.